py/torch_tensorrt/dynamo/conversion/impl/slice/ops.py

```diff
@@ -90,27 +92,39 @@ def expand(
     # After the above padding, the shape and tensor rank must be equal
     assert len(input_t.shape) == shape_rank
 
-    # -1 denotes taking the shape from the original input tensor
-    shape = tuple(
-        [input_t.shape[i] if shape[i] == -1 else shape[i] for i in range(shape_rank)]
-    )
+    shape_t = []
+    for i in range(shape_rank):
+        if shape[i] == -1:
+            shape_t.append(
+                get_shape(ctx, target, source_ir, name + f"_shape_dim{i}", input_t, i)
+            )
+        else:
+            shape_t.append(shape[i])
 
     # Establish the desired output shape, strides, and starting indices
     input_tensor_shape = tuple(input_t.shape)
     start = tuple([0] * shape_rank)
-    stride = tuple(
-        [int(i == o) for i, o in zip(input_tensor_shape, shape)]
-    )  # stride == 1 if dimensions match, 0 otherwise
 
-    shape_ = shape
+    # TODO: Revisit stride calculation. stride[dim]=0 implies that dimension is being broadcasted.
+    # stride should be 1 for all non-broadcasted dims
+    stride = []
+    for i, o in zip(input_tensor_shape, shape_t):
+        # If the shape has ITensor, we treat it as a reshape dim instead of a broadcasted dim
+        # shape_t cannot have -1. If the input at this dimension has a shape of -1, set the stride to 1. This indicates that the input is dynamic and does not imply broadcasting at that specific dimension.
+        if isinstance(i, int) and isinstance(o, int) and i != DYNAMIC_DIM:
+            stride.append(int(i == o))
+        else:
```

Comment:
Is the check for i != DYNAMIC_DIM necessary here? Isn't DYNAMIC_DIM = -1 , but say if the expanded shape at the dimension dim is -1, and the input tensor shape at dim is also -1, shouldn't the shape appended here 
```
shape_t.append(
                get_shape(ctx, target, source_ir, name + f"_shape_dim{i}", input_t, i)
            )
```
be a shape tensor and not -1?

Reply:
`shape_t` won't have a -1 since we convert them into ITensors. Let's say there is a particular `dim` in input_tensor which has shape of -1. But the shape_t has a defined value for it (eg: 2). int(i==0) will return 0 and results in broadcasting which shouldn't happen.  For eg: this situation happens if the 5th testcase is modified as `("different_ranks", (1, 2, 1), (1, 2, 1), (2, 2, 1), (2, 2, -1, -1))`. Hence, if we find a dimension of `-1` in the input_tensor the stride should be 1 for that location.

Reply:
Yes makes sense. I missed the `isinstance(i, int)`  which can have -1.

Reply:
Yes ok, I missed the `isinstance(i, int)` which can be -1 in input tensor.